abi/abi-0.2.ts

```diff
@@ -0,0 +1,171 @@
+/// ABIs
+
+export interface AbiDefs {
+  functions?: FunctionDef[];
+  objects?: ObjectDef[];
```

Comment:
Calling this typeDefinition sounds more intuitive to me.

Reply:
I am 50/50 on this but it feels right to call it type definition because it corresponds to type in the SDL

Reply:
Well, one of the things we wanted to do with the `clearer semantics` point was to easily identify which types are built-in, or refs, and which are user defined; so we're calling the former "types" and the latter "definitions". So "type definitions" sounds like it brings back the ambiguity to me. But I don't hold a strong opinion on this if you think this should be different. Thoughts @dOrgJelli @Niraj-Kamdar ?

Reply:
Yes, "type definition" is simple too ambiguous @Niraj-Kamdar, please read the "clearer semantics" section in our readme doc that covers this decision made.

Reply:
Yes, "type definition" is simple too ambiguous @Niraj-Kamdar, please read the "clearer semantics" section in our readme doc that covers this decision made.